src/Control/Monad/Log.hs

```diff
@@ -840,29 +840,29 @@ For example, we've already seen the basic @testApp@ computation above that used
 have some older code that doesn't yet have any severity information:
 
 @
-legacyCode :: 'MonadLog' 'PP.Doc' m => m ()
+legacyCode :: 'MonadLog' ('PP.Doc' ann) m => m ()
 legacyCode = 'logMessage' "Does anyone even remember writing this function?"
 @
 
 Here @legacyCode@ is only logging 'PP.Doc', while our @testApp@ is logging
 'WithSeverity' 'PP.Doc'. What happens if we compose these programs?
 
->>> :t testApp >> legacyCode
-  Couldn't match type ‘Doc’ with ‘WithSeverity Doc’
+>>> :t runLoggingT (testApp >> legacyCode) (const (pure ()))
```

Comment:
I changed this example in the next one to include `runLoggingT` instead of just `testApp >> legacyCode` because without, ghci seemed to be having a hard time inferring the monad constraint from using `(>>)`, and the type variable on the `Doc` type was ambiguous:

```
<interactive>:1:1: error:
    • Ambiguous type variable ‘ann0’ arising from a use of ‘testApp’
      prevents the constraint ‘(MonadLog
                                  (WithSeverity (PP.Doc ann0)) m)’ from being solved.
      Probable fix: use a type annotation to specify what ‘ann0’ should be.
      These potential instances exist:
        instance Monad m => MonadLog message (DiscardLoggingT message m)
          -- Defined in ‘Control.Monad.Log’
        instance Monad m => MonadLog message (LoggingT message m)
          -- Defined in ‘Control.Monad.Log’
        instance (Monad m, Monoid log) => MonadLog log (PureLoggingT log m)
          -- Defined in ‘Control.Monad.Log’
        ...plus 16 instances involving out-of-scope types
        (use -fprint-potential-instances to see them all)
    • In the first argument of ‘(>>)’, namely ‘testApp’
      In the expression: testApp >> legacyCode

<interactive>:1:1: error:
    • No instance for (Monad m) arising from a use of ‘>>’
      Possible fix:
        add (Monad m) to the context of the inferred type of it :: m ()
    • In the expression: testApp >> legacyCode

<interactive>:1:12: error:
    • Ambiguous type variable ‘ann1’ arising from a use of ‘legacyCode’
      prevents the constraint ‘(MonadLog
                                  (PP.Doc ann1) m)’ from being solved.
      Probable fix: use a type annotation to specify what ‘ann1’ should be.
      These potential instances exist:
        instance Monad m => MonadLog message (DiscardLoggingT message m)
          -- Defined in ‘Control.Monad.Log’
        instance Monad m => MonadLog message (LoggingT message m)
          -- Defined in ‘Control.Monad.Log’
        instance (Monad m, Monoid log) => MonadLog log (PureLoggingT log m)
          -- Defined in ‘Control.Monad.Log’
        ...plus 16 instances involving out-of-scope types
        (use -fprint-potential-instances to see them all)
    • In the second argument of ‘(>>)’, namely ‘legacyCode’
      In the expression: testApp >> legacyCode
```